test/reflection.jl

```diff
@@ -809,3 +809,6 @@ f20872(::Val, ::Val) = false
 @test which(f20872, Tuple{Val,Val}).sig == Tuple{typeof(f20872), Val, Val}
 @test which(f20872, Tuple{Val,Val{N}} where N).sig == Tuple{typeof(f20872), Val, Val}
 @test_throws ErrorException which(f20872, Tuple{Any,Val{N}} where N)
+
+# make sure checking if a binding is deprecated does not resolve it
+@test !(!Base.isdeprecated(Main, :sin) && !Base.isbindingresolved(Main, :sin))
```

Comment:
I recommend making an empty module for this purpose so it doesn't depend on `Main`, which will be unpredictable.